### PR TITLE
[EM-1249] add global kpi endpoint on phoning campaign

### PR DESCRIPTION
**Pull request by elvino29**

### CODE REVIEW

File: src/Repository/Phoning/CampaignRepository.php [hunk 1]
```diff
@@ -47,4 +47,30 @@ public function findForAdherent(Adherent $adherent): array
...
+            ->leftJoin(
+                'campaign.campaignHistories',
+                'campaignHistory',
+                Join::WITH,
+                'campaignHistory.status != :send'
+            )
+            ->leftJoin('campaignHistory.dataSurvey', 'dataSurvey')
+            ->setParameters([
+                'end' => new \DateTime(),
+                'start' => new \DateTime('last month'),
```
Comment (illusionOfParadise) [inline src/Repository/Phoning/CampaignRepository.php:69]: What about names corresponding to the date, to see immediately with what we compare?
```suggestion
                'now' => new \DateTime(),
                'last_month' => new \DateTime('last month'),
```

File: src/Repository/Phoning/CampaignRepository.php [hunk 2]
```diff
@@ -47,4 +47,30 @@ public function findForAdherent(Adherent $adherent): array
             ->getResult()
         ;
     }
+
+    public function findPhoningCampaignsKpi()
+    {
+        return $this->createQueryBuilder('campaign')
+            ->select('COUNT(DISTINCT campaign.id) AS nb_campaigns')
+            ->addSelect('COUNT(DISTINCT IF(campaign.finishAt >= :now OR campaign.finishAt IS NULL, campaign.id, null)) AS nb_on_going_campaigns')
```
Comment (illusionOfParadise) [inline src/Repository/Phoning/CampaignRepository.php:55]: Same here.
```suggestion
            ->addSelect('COUNT(DISTINCT IF(campaign.finishAt >= :now OR campaign.finishAt IS NULL, campaign.id, null)) AS nb_ongoing_campaigns')
```

File: src/Controller/Api/Phoning/GetPhoningCampaignsKpiController.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Controller\Api\Phoning;
+
+use App\Repository\Phoning\CampaignRepository;
+
+class GetPhoningCampaignsKpiController
+{
+    public function __invoke(CampaignRepository $campaignRepository)
```
Comment (ottaviano) [inline src/Controller/Api/Phoning/GetPhoningCampaignsKpiController.php:9]: return typehint missing

File: src/Repository/Phoning/CampaignRepository.php [hunk 4]
```diff
@@ -47,4 +47,30 @@ public function findForAdherent(Adherent $adherent): array
             ->getResult()
         ;
     }
+
+    public function findPhoningCampaignsKpi()
```
Comment (ottaviano) [inline src/Repository/Phoning/CampaignRepository.php:51]: typehint missing

File: src/Repository/Phoning/CampaignRepository.php [hunk 5]
```diff
@@ -47,4 +47,30 @@ public function findForAdherent(Adherent $adherent): array
...
+    {
+        return $this->createQueryBuilder('campaign')
+            ->select('COUNT(DISTINCT campaign.id) AS nb_campaigns')
+            ->addSelect('COUNT(DISTINCT IF(campaign.finishAt >= :now OR campaign.finishAt IS NULL, campaign.id, null)) AS nb_ongoing_campaigns')
+            ->addSelect('COUNT(campaignHistory.id) AS nb_calls')
+            ->addSelect('SUM(IF(campaignHistory.beginAt >= :last_month AND campaignHistory.beginAt <= :now, 1, 0)) AS nb_calls_last_month')
+            ->addSelect('SUM(IF(campaignHistory.dataSurvey IS NOT NULL, 1, 0)) as nb_surveys')
```
Comment (ottaviano) [inline src/Repository/Phoning/CampaignRepository.php:58]:
```suggestion
            ->addSelect('COUNT(campaignHistory.dataSurvey) as nb_surveys')
```

**Review by ottaviano — APPROVED** [review]

**Review by ottaviano — APPROVED** [review]

**Pull request merged**